frameworks/react/__tests__/AdvancedPicture.test.tsx

```diff
@@ -98,10 +101,183 @@ describe('AdvancedPicture', () => {
     setTimeout(() => {
       expect(component.html()).toContain(
         '<picture>' +
+        '<img src="https://res.cloudinary.com/demo/image/upload/sample"><source media="(max-width: 500px)" sizes="80vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog"></picture>');
+      done();
+    }, 0);// one tick
+  });
+
+  it('autoOptimalBreakpoints: should generate default srcset when min-width and max-width not provided',
+    function (done) {
+      const component = mount(
+        <AdvancedPicture
+          autoOptimalBreakpoints
+          cldImg={defualtImage}
+          sources={[
+            {
+              image: sepiaImage
```

Comment:
As far as I understand the spec right, at least one of min or max is required. This is an invalid state.

Reply:
@const-cloudinary 
See the following 
Srcset generation flow
If user didn’t pass min-width or max-width: set the missing value to its default.

and the first example #1 - empty inputs

Reply:
that relates to the `Srcset generation flow`, which is `srcset` attribute, not `media` attribute. for `media` attribute at least one of them is mandatory